src/test/test-mountpoint-util.c

```diff
@@ -287,6 +290,27 @@ static void test_fd_is_mount_point(void) {
         assert_se(IN_SET(fd_is_mount_point(fd, "root/", 0), -ENOENT, 0));
 }
 
+static void test_make_mount_point_inode(void) {
+        _cleanup_(rm_rf_physical_and_freep) char *d = NULL;
+        struct stat st;
+
+        log_info("/* %s */", __func__);
+
+        assert_se(mkdtemp_malloc(NULL, &d) >= 0);
+        assert_se(mkdir_p(strjoina(d, "/src/dir"), 0755) >= 0);
+        assert_se(touch(strjoina(d, "/src/file")) >= 0);
+        assert_se(make_mount_point_inode(strjoina(d, "/src/dir"), strjoina(d, "/dst/dir"), 0755) >= 0);
+        assert_se(make_mount_point_inode(strjoina(d, "/src/file"), strjoina(d, "/dst/file"), 0755) >= 0);
```

Comment:
this is problematic. strjoina() allocates on the stack, and function params are placed on the stack too. doing alloca() inside of function param lists is hence at least traditionally quite problematic, since it fucks up the stack actually passed. gcc might handle this OK these days, dunno, but we probably still shouldn't do it. i.e. do the strjoina() in separate lines, beforehand

Reply:
Yeah, I avoid that now in the "main" code, but I thought for small unit tests binaries it would be fine, as it seems to work without issues. I'll rework it.